File: src/main/java/net/pterodactylus/rhynodge/filters/comics/BusinessCatComicFilter.java
package net.pterodactylus.rhynodge.filters.comics;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import net.pterodactylus.rhynodge.filters.ComicSiteFilter;

import com.google.common.base.Optional;
import org.jsoup.nodes.Document;

/**
 * {@link ComicSiteFilter} implementation that can parse Business Cat comics.
 *
 * @author <a href="mailto:bombe@pterodactylus.net">David ‘Bombe’ Roden</a>
 */
public class BusinessCatComicFilter extends ComicSiteFilter {

	@Override
	protected Optional<String> extractTitle(Document document) {
		String title = document.select("#comic img").attr("title");
		return title.isEmpty() ? Optional.<String>absent() : Optional.of(title);
	}

	@Override
	protected List<String> extractImageUrls(Document document) {
		return Arrays.asList(document.select("#comic img").attr("src"));
	}

	@Override
	protected List<String> extractImageComments(Document document) {
		return Collections.emptyList();
	}

}

File: src/main/kotlin/net/pterodactylus/rhynodge/filters/comics/BusinessCatComicFilter.kt
package net.pterodactylus.rhynodge.filters.comics

import com.google.common.base.Optional
import net.pterodactylus.rhynodge.filters.ComicSiteFilter
import net.pterodactylus.rhynodge.utils.asOptional
import org.jsoup.nodes.Document

class BusinessCatComicFilter : ComicSiteFilter() {

	override fun extractTitle(document: Document): Optional<String> =
			document.select(".comic-title h2").text().asOptional()

	override fun extractImageUrls(document: Document): List<String> =
			listOf(document.select(".comic-image img").attr("src"))

	override fun extractImageComments(document: Document): List<String> = emptyList()

}
